### PR TITLE
HTS-1041 Adds close account page

### PR DESCRIPTION
**Pull request by thebailers**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/controllers/AccountHolderController.scala [hunk 1]
```diff
@@ -133,7 +133,13 @@ class AccountHolderUpdateEmailAddressController @Inject() (val helpToSaveService
       },
       email ⇒ Ok(views.html.email.we_updated_your_email(email))
     )
-  }(redirectOnLoginURL = routes.AccountHolderUpdateEmailAddressController.getEmailVerified.url)
+  }(redirectOnLoginURL = routes.AccountHolderController.getEmailVerified.url)
+
+  def getCloseAccountPage: Action[AnyContent] = authorisedForHtsWithNINO { implicit request ⇒ implicit htsContext ⇒
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/AccountHolderController.scala]: do we need `authorisedForHtsWithNINO ` can we use `authorisedForHts` instead here ?

File: app/uk/gov/hmrc/helptosavefrontend/views/close_account_are_you_sure.scala.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.helptosavefrontend.models.HtsContext
+@import uk.gov.hmrc.helptosavefrontend.config.FrontendAppConfig
+
+@()(implicit htsContext: HtsContext, request: Request[_], messages: Messages, appConfig: FrontendAppConfig)
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/views/close_account_are_you_sure.scala.html]: do we need `htsContext ` in this page ?

Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosavefrontend/views/close_account_are_you_sure.scala.html]: yea, we need it, please ignore above message

File: app/uk/gov/hmrc/helptosavefrontend/controllers/AccountHolderController.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/AccountHolderController.scala]: I am using authorisedForHtsWithNINO so I can checkIfAlreadyEnrolled, is this check necessary, will a user who is not yet enrolled be able to access this page?

File: app/uk/gov/hmrc/helptosavefrontend/controllers/AccessAccountController.scala [hunk 3]
```diff
@@ -59,7 +59,20 @@ class AccessAccountController @Inject() (val helpToSaveService: HelpToSaveServic
...
   }(redirectOnLoginURL = routes.AccessAccountController.getNoAccountPage().url)
 
+  def getCloseAccountPage: Action[AnyContent] = authorisedForHtsWithNINO { implicit request ⇒ implicit htsContext ⇒
+    checkIfEnrolledForCloseAccount(
+      { // not enrolled
+        () ⇒ SeeOther(routes.AccessAccountController.getNoAccountPage().url)
+      }, {
+        e ⇒
+          logger.warn(s"Could not check enrolment during call to close account page ($e)", htsContext.nino)
+          InternalServerError
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/AccessAccountController.scala]: you should use `internalServerError()` instead

File: app/uk/gov/hmrc/helptosavefrontend/controllers/AccessAccountController.scala [hunk 4]
```diff
@@ -70,7 +70,7 @@ class AccessAccountController @Inject() (val helpToSaveService: HelpToSaveServic
       }, {
         e ⇒
           logger.warn(s"Could not check enrolment during call to close account page ($e)", htsContext.nino)
-          InternalServerError
+          internalServerError
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/AccessAccountController.scala]: i was expecting `()` but if its comping fine , then no issues

Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/AccessAccountController.scala]: yeah its compiling

**Pull request merged**